### PR TITLE
Spanish translation of the docs

### PR DESCRIPTION
**Pull request by LautaroJayat**
+ Added Spanish translations for all the index.md files in the `/docs/`
+ Added a link at the bottom of the index.md file inside the root folder.
+ The translation will also be revised for further improvements

I think it's ready to merge.

### CODE REVIEW

**Comment by EvanHahn** [comment]
¡Gracias! I will look at this soon.

**Comment by LautaroJayat** [comment]
Cool!
Let me know if you wan't any correction, or extra information, or something ;)

File: index.md [hunk 1]
```diff
@@ -44,3 +44,8 @@ You can see more in [the documentation](/docs).
 | [permittedCrossDomainPolicies](/docs/crossdomain/) for handling Adobe products' crossdomain requests |  |
 | [referrerPolicy](/docs/referrer-policy) to hide the Referer header |  |
 | [xssFilter](/docs/xss-filter) adds some small XSS protections | ✓ |
+
+
+To see the documentation in other languages
```
Comment (EvanHahn) [inline index.md:49]: Could you remove this? I will link to Spanish documentation in a separate commit.

File: translations/es/crossdomain/index.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+title: X-Permitted-Cross-Domain-Policies
+permalink: /translations/es/crossdomain/
+---
+Brevemente: La función *middleware* `crossdomain` de Helmet previene que Adobe Flash y Adobe Acrobat carguen contenido en tu sitio.
+
+El "ataque"
+------------
+Limitar el área superficial de tu código es una buena idea. Menos cosas implican menos ataques.
+
+Adobe Flash y Adobe Acrobat pueden cargar contenido en tu dominio incluso desde otros sitios (en otras palabtas, "*cross-domain*"). Esto podría causar inesperadas filtraciones de datos o un uso excesivo del ancho de banda.
```
Comment (EvanHahn) [inline translations/es/crossdomain/index.md:12]: *palabtas* → *palabras*?

File: translations/es/crossdomain/index.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+La cabecera
+----------
+
+La cabecera `X-Permitted-Cross-Domain-Policies` dice a los clientes Flash and Acrobat qué políticas de dominio cruzado ("*cross-domain*) pueden usar. Si no quieres que carguen contenido desde tu dominio, setea el valor de ésta cabecera a `none`. Por Ejemplo:
```
Comment (EvanHahn) [inline translations/es/crossdomain/index.md:17]: *Ejemplo* → *ejemplo*?

File: translations/es/crossdomain/index.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+```
+X-Permitted-Cross-Domain-Policies: none
+```
+Si flash carga algo desde tu sitio y ve eso, entonces sabrá que no debe cargar contenido desde tu dominio.
```
Comment (EvanHahn) [inline translations/es/crossdomain/index.md:22]: *flash* → *Flash*?

File: translations/es/crossdomain/index.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+La cabecera posee otros valores que requieren la creación de un archivo `crossdomain.xml` que defina cuáles son tus políticas de dominio cruzado (*"cross-domain"*). Puedes leer más al respecto en los enlaces de abajo.
+
+Para más información:
```
Comment (EvanHahn) [inline translations/es/crossdomain/index.md:26]: ¿Deberíamos mencionar que estos enlaces son en íngles? (No sé.)

File: translations/es/csp/index.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
+---
+layout: page
+title: Content Security Policy
+permalink: /translations/es/csp/
+---
+Brevemente: el módulo CSP setea la cabecera `Content-Security-Policy`, la cual puede ayudar a proteger contra inyecciones maliciosas de Javascript, CSS, *pluguins*, y más.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:6]: 1. *Javascript* → *JavaScript*? This needs to be fixed everywhere.
1. *pluguins* → *plugins*?

File: translations/es/csp/index.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+El ataque
+----------
+Los hackers pueden realizar numerosas maldades si logran poner cosas en tus páginas.
+
+El ataque más sucio es probablemente el *cross-site scripting* (XSS), el cual ocurre cuando un hacker pone código malicioso en tu sitio. Si yo puedo ejecutar Javascript en tu página, entonces puedo hacer cosas malas ahí, desde robar cookies de autenticación hasta registrar las acciones de todos los usuarios.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:12]: ¿Deberíamos poner en cursiva "*cookies*"?

File: translations/es/csp/index.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+Existen otras cosas que los atacantes pueden hacer, incluso si no pueden ejecutar Javascript. por ejemplo, si uno pudiera poner una pequeña imagen en tu sitio, 1x1 y transparente, podría tener una buena idea de cuánto tráfico llega a tu sitio. Si uno pudiera lograr que el navegador corriera un [pluguin vulnerable](http://arstechnica.com/security/2015/07/two-new-flash-exploits-surface-from-hacking-team-combine-with-java-0-day/) como flash, entonces podría explotar sus falencias y realizar acciones que desearías evitar.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:14]: 1. "por ejemplo" → "Por ejemplo"
1. `http` → `https`. (Necesito hacerlo en la documentación en íngles, tambien. Hice #47.)
1. "flash" → "Flash"

File: translations/es/csp/index.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+
+El módulo CSP no proteje contra un ataque específico. Aquí lo principal es que no quieres a nadie poniendo nada inesperado dentro de tus sitios.
+
+Para más información:
+
+* [Cross-site scripting en Wikipedia](https://en.wikipedia.org/wiki/Cross-site_scripting)
+* [Cross-site Scripting en OWASP](https://www.owasp.org/index.php/Cross-site_Scripting_%28XSS%29)
+* ["How does a tracking pixel work?" en Quora](https://www.quora.com/How-does-a-tracking-pixel-work)
+
+La Cabecera
```
Comment (EvanHahn) [inline translations/es/csp/index.md:24]: "Cabecera" → "cabecera"

File: translations/es/csp/index.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+----------
+
+Una de las complicaciones acerca de éstas ataques de inyección de código es que los navegadores no saben qué es bueno y qué es malo. ¿Cómo podría diferenciar entre un archivo de Javascript legítimo y uno malicioso? En muchos casos, no puede... a menos que yu hayas definido *"Content Security Policy*.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:27]: 1. ¿"yu" es un error de tipografía?
1. Hay un extra `"` aquí.

File: translations/es/csp/index.md [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+* [Cross-site scripting en Wikipedia](https://en.wikipedia.org/wiki/Cross-site_scripting)
+* [Cross-site Scripting en OWASP](https://www.owasp.org/index.php/Cross-site_Scripting_%28XSS%29)
+* ["How does a tracking pixel work?" en Quora](https://www.quora.com/How-does-a-tracking-pixel-work)
+
+La Cabecera
+----------
+
+Una de las complicaciones acerca de éstas ataques de inyección de código es que los navegadores no saben qué es bueno y qué es malo. ¿Cómo podría diferenciar entre un archivo de Javascript legítimo y uno malicioso? En muchos casos, no puede... a menos que yu hayas definido *"Content Security Policy*. 
+
+La mayoría de los navegadores modernos soportan la cabecera llamada `Content-Security-Policy`, la cual efectivamente es una lista blanca de las cosas que tienen permitido estar en tu página. Podes permitir Javascript, CSS, imágenes, *pluguins*, y mucho más. Aquí las cosas están "*opt-in*", lo que significa que creas una lista de "**cosas que están permitidas**", en vez de crear una lista de **cosas que NO están permitidas**.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:29]: *pluguins* → *plugins*?

File: translations/es/csp/index.md [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+
+Esto efectivamente le dice al navegador "sólo carga cosas que sean de mi propio dominio". Si estás corriendo *ejemplo.com* y un usuario trata de cargar `https://ejemplo.com/mi-javascript.js`, entonces todo va a ejecutarse bien. ¡Pero si un usuario trata de ejecutar `https://maligno.com/maligno.js`, entonces nada va a cargarse!
+
+Ahora, digamos que sí quieres permitir la carga de CSS de Bootstrap desde su CDN. Entonces podrías setear una CSP que luzca como ésta:
+
+```
+Content-Security-Policy: default-src 'self'; style-src 'self' maxcdn.bootstrapcdn.com
+```
+
+Ahora `'self'` y `maxcdn.bootstrapcdn.com` en nuestra lista blanca. los usuarios podrán cargar CSS desde ahí, pero desde ningún otro lugar. Los usuarios ni siquiera tendrán permitido cargar Javascriot o imágenes desde la nueva CDN, sólamente hojas de estilo.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:45]: 1. "en nuestra lista blanca" → "están en nuestra lista blanca"?
1. "los usarios" → "Los usarios"
1. "Javascriot" → "JavaScript"

File: translations/es/csp/index.md [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+
+```js
+// Necesitarás JSON parser primero.
+app.use(bodyParser.json({
+  type: ['json', 'application/csp-report']
+}))
+
+app.post('/report-violation', (req, res) => {
+  if (req.body) {
+    console.log('CSP Violation: ', req.body)
```
Comment (EvanHahn) [inline translations/es/csp/index.md:149]: ¿Deberíamos traducir estos `console.log`s?

File: translations/es/csp/index.md [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+
+```javascript
+app.use(csp({
+  directives: {
+    // ...
+  },
+  reportOnly: true
+})
+```
+Tambén puedes asignar esto a una función para decidir de manera dinámica cuándo usar el modo `reportOnly`. Puedes usar ésto como un disyuntor dinámico. En éste caso, la función es llamada usando los objetos "*request*" y "*response*" y debe retornar una variable de tipo *Boolean*.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:172]: *Tambén* → *También*?

File: translations/es/csp/index.md [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+```javascript
+app.use(csp({
+  directives: {
+    // ...
+  },
+  reportOnly: (req, res) => req.query.cspmode === 'debug'
+})
+```
+
+### Browser sniffing (detección de navegadores).
```
Comment (EvanHahn) [inline translations/es/csp/index.md:183]: 1. ¿Se debe poner en cursiva "Browser sniffing"?
1. Por favor, elimine el punto.

File: translations/es/csp/index.md [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+
+Por defecto, éste módulo se fijará en la cabecera `User-Agent` entrante y enviará diferentes cabeceras dependiendo en el navegador detectado. por ejemplo, las versiones previas a Chrome 25 usan una cabecera alternativa llamada `X-WebKit-CSP`, el cuál también es manejado por éste módulo. Si ningún navegador es detectado, entonces las cabeceras serán seteadas con las especificaciones 2.0.
```
Comment (EvanHahn) [inline translations/es/csp/index.md:185]: "por ejemplo" → "Por ejemplo"

File: translations/es/dns-prefetch-control/index.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
+---
+layout: page
+title: DNS Prefetch Control
+permalink: /docs/dns-prefetch-control/
+---
+In short: this middleware lets you disable browsers' DNS prefetching by setting the `X-DNS-Prefetch-Control` header.
```
Comment (EvanHahn) [inline translations/es/dns-prefetch-control/index.md:6]: In English: this file was not translated. Could we translate it (or delete most of it and add a link to [the English docs][0])?

En español: este archivo no está traducido. Quiero que lo traduces (o elimines el contento y añades un elace al [version íngles][0]), por favor.

[0]: https://helmetjs.github.io/docs/dns-prefetch-control/

File: translations/es/dont-sniff-mimetype/index.md [hunk 18]
```diff
@@ -0,0 +1,71 @@
+---
+layout: page
+title: Don't Sniff Mimetype
+permalink: /translations/es/dont-sniff-mimetype/
+---
+
+Brevemente: la funcción middleware "*Don't Sniff Mimetype middleware*", `noSniff`, ayuda a prevenir que los navegadores intenten inferir los *MIME type*, lo que puede tener implicancias en la seguridad. esto se realiza seteando la cabecera `X-Content-Type-Options` en `nosniff`.
```
Comment (EvanHahn) [inline translations/es/dont-sniff-mimetype/index.md:7]: 1. `la funcción middleware "*Don't Sniff Mimetype middleware*"` → `la funcción middleware "*Don't Sniff Mimetype*"`
1. "esto se realiza" → "Esto se realiza"

File: translations/es/dont-sniff-mimetype/index.md [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+```
+En éste caso, el navegador cargará `my-javascript` from `example.com`. Si `example.com` envía una cabecera `Content-Type` con el contenido `text/javascript`, el navegador ejecutará los contenidos de `my-javascript` como Javascript.
+
+Pero que pasaría si `my-javascript` es una página HTML con un `Content-Type` seteado en `text/html`? Si tu navegador hace algo llamado "*MIME sniffing*" (algunos lo hacen), entonces se fijará en los contenidos de dicho archivo, decidirá si se parece a Javascript, y posteriormente lo ejecutará si así lo interpreta. Esto significa que un servidor puede enviar un `Content-Type` equivocado y que el contenido Javascript se ejecute de todas formas.
+
+Este *Mime sniffing* puede ser un vector de ataques. un usuario podría subir una archivo con una extensión `.jpg`, pero con HTML en su contenido. Cargar esa página podría causar que el navegador "ejecute" la página HTML dentro del archivo... ¡El cuál podría contener Javascript malicioso! Quizás el ataque más sucio es conocido como [Rosetta Flash](https://miki.it/blog/2014/7/8/abusing-jsonp-with-rosetta-flash/), el cual permite que alguin logre cargar un *pluguin* malicioso de Flash.
```
Comment (EvanHahn) [inline translations/es/dont-sniff-mimetype/index.md:23]: 1. "Mime" → "MIME"
1. "un usuario" → "Un usuario"
1. "pluguin" → "plugin"

File: translations/es/dont-sniff-mimetype/index.md [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+Para más información:
+
+- ["MIME type" on Wikipedia](https://en.wikipedia.org/wiki/Media_type)
+- ["MIME Sniffing: feature or vulnerability?"](https://blog.fox-it.com/2012/05/08/mime-sniffing-feature-or-vulnerability/)
+- ["MIME sniffing in Internet Explorer enables cross-site scripting attacks"](http://www.h-online.com/security/features/Risky-MIME-sniffing-in-Internet-Explorer-746229.html)
+- ["Abusing JSONP with Rosetta Flash"](https://miki.it/blog/2014/7/8/abusing-jsonp-with-rosetta-flash/)
+- [MIME Sniffing live standard](https://mimesniff.spec.whatwg.org/)
+- [Cross Origin Resource Blocking in Chrome](https://developers.google.com/web/updates/2018/07/site-isolation)
+
+La Cabecera
```
Comment (EvanHahn) [inline translations/es/dont-sniff-mimetype/index.md:34]: "Cabecera" → "cabecera"

File: translations/es/dont-sniff-mimetype/index.md [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+app.use(helmet.noSniff())
+```
+
+También puedes usar éste módulo en solitario:
+
+```javascript
+// Asegurate de haber ejecutado "npm install dont-sniff-mimetype" para obtener éste paquete.
+const noSniff = require('dont-sniff-mimetype')
+
+// Asigna"X-Content-Type-Options: nosniff".
```
Comment (EvanHahn) [inline translations/es/dont-sniff-mimetype/index.md:67]:
```diff
-// Asigna"X-Content-Type-Options: nosniff".
+// Asigna "X-Content-Type-Options: nosniff".
```

File: translations/es/expect-ct/index.md [hunk 22]
```diff
@@ -0,0 +1,28 @@
+---
+layout: page
+title: Expect-CT
+permalink: /translations/es/expect-ct/
+---
+La cabecera HTTP `Expect-CT` indica a los navegadores que deben esperar ("*Certificate Transparency*"). Para saber más acerca de *Certificate Transparency y ésta cabecera, ve [este artículo](https://scotthelme.co.uk/a-new-security-header-expect-ct/) y esta [especificación](https://datatracker.ietf.org/doc/draft-stark-expect-ct).
```
Comment (EvanHahn) [inline translations/es/expect-ct/index.md:6]: Necesitamos un `*` después de la palabra "Transparency".

```diff
-Para saber más acerca de *Certificate Transparency y ésta cabecera
+Para saber más acerca de *Certificate Transparency* y ésta cabecera
````

**Review by EvanHahn — CHANGES_REQUESTED** [review]
In English: Thanks so much! I have reviewed a few files but there's still more for me to look at. Could you fix these comments while I review the rest?

En español: ¡Muchas gracias! Revisé algunos de los archivos pero todavía hay más que tengo revisar. Arregles mis commentos, por favor.

**Comment by LautaroJayat** [comment]
Sorry pal!
I will correct all of this as soon as possible!

**Comment by EvanHahn** [comment]
I plan to release Helmet version 4 on 2020-08-02. In that release, I will be removing large sections of the documentation (see #48). That means that this pull request can be _much_ smaller and easier for you to maintain. I wanted to let you know!

I will leave another comment next week after the release is finished.

**Comment by LautaroJayat** [comment]
@EvanHahn sorry about not making the requested changes, I have been extra busy at work.
I promise that I will read the new documentation and try to translate it. And also make the requested changes so the current version can be translated too.

**Comment by EvanHahn** [comment]
No problem! Take your time.

**Comment by EvanHahn** [comment]
Helmet 4 was just released, so there should be a lot less work. Thank you, and take your time!

**Comment by EvanHahn** [comment]
It's been 2 years since this PR was made. I'm going to close it. I would like to have Spanish translations of the docs, so feel free to make a new one!

**Pull request closed without merging**